plugin/CactbotEventSource/FateWatcher.cs

```diff
@@ -132,26 +124,96 @@ public struct CEDirectorData {
+      CEOpcodeList localOpcodeList = new CEOpcodeList();
+
+      if (!Config.DisableAutomaticOpcodeUpdates) {
+
+        CEOpcodeList remoteOpcodeList = new CEOpcodeList();
+        string remoteOpcodeJSON = null;
+
+        try {
+          // Do update check
+          if (Config.UseCustomOpcodeSource && Config.CustomOpcodeSourceUrl != null) {
+            remoteOpcodeJSON = GetRemoteOpcodeJSON(Config.CustomOpcodeSourceUrl);
+
+          // Default url
+          } else {
+            remoteOpcodeJSON = GetRemoteOpcodeJSON(remoteOpcodeURL);
+          }
+        } catch (HttpRequestException ex) {
+          client_.LogError("Error fetching remote opcodes, using local cache (may be outdated).\r\n" +
+            "Exception: {0}", ex.Message);
+
+          if (Config.UseCustomOpcodeSource) {
+            client_.LogWarning("The custom url may be invalid.\r\n" +
+              "Custom URL: {0}", Config.CustomOpcodeSourceUrl);
+          }
+        };
+        remoteOpcodeList = JsonConvert.DeserializeObject<CEOpcodeList>(remoteOpcodeJSON,
+          new JsonSerializerSettings {
+            Error = delegate (object sender, Newtonsoft.Json.Serialization.ErrorEventArgs args) {
+              errors.Add(args.ErrorContext.Error.Message);
+              args.ErrorContext.Handled = true;
+            }
+          });
+
+        // Successfully got remote opcodes
+        if (errors.Count == 0) {
+
+          if (Config.OverlayData["options"]["opcodes"] != null) {
+            localOpcodeList = JsonConvert.DeserializeObject<CEOpcodeList>(Config.OverlayData["options"]["opcodes"].ToString());
+          }
+          if (Config.OverlayData["options"]["opcodes"] == null || remoteOpcodeList.version > localOpcodeList.version) {
+            // update local storage
+            Config.OverlayData["options"]["opcodes"] = JToken.Parse(JsonConvert.SerializeObject(remoteOpcodeList));
+          }
+
+        } else {
+          client_.LogError("There were errors encountered parsing the remote opcodes, the local cache has not been updated:\r\n" +
+            "{0}", String.Join("\n", errors));
+        }
+        errors.Clear();
+        // We finished the update check and updated our local file if necessary
+      }
+
+      localOpcodeList = JsonConvert.DeserializeObject<CEOpcodeList>(Config.OverlayData["options"]["opcodes"].ToString(),
+          new JsonSerializerSettings {
+            Error = delegate (object sender, Newtonsoft.Json.Serialization.ErrorEventArgs args) {
+              errors.Add(args.ErrorContext.Error.Message);
+              args.ErrorContext.Handled = true;
+            }
+          });
+
+      if (errors.Count > 0) {
+        client_.LogError("There were errors encountered parsing the local opcodes:\r\n" +
+          "{0}" +
+          "FATE/CE data is unavailable.", String.Join("\n", errors));
+        return;
+      }
+
       if (language == "ko")
-        region_ = "ko";
+        OpCodes = localOpcodeList.ko;
```

Comment:
Ok, I know cactbot is not consistent about capitalization, but can I request lowercase camelCase for variables and CapitalCamelCase for class names for some attempt at consistency?